test/unit/services/asset_manager/attachment_updater/access_limited_updates_test.rb

```diff
@@ -8,89 +8,189 @@ class AssetManager::AttachmentUpdater::AccessLimitedUpdatesTest < ActiveSupport:
     let(:attachment_data) { attachment.attachment_data }
     let(:update_worker) { mock("asset-manager-update-asset-worker") }
 
-    around do |test|
-      AssetManager.stub_const(:AssetUpdater, update_worker) do
-        test.call
+    describe "Attachment Data has no assets" do
```

Comment:
For this test and other similar to it, we have split the tests in Feature flag on and off respectively (in our case: assets present or not). We have then replicated all tests to match the new implementation and on occasion added more tests or slightly adjusted the logic. 
The Asset model simplifies the tests a little bit in that there is no need to always tests whether the attachment is a pdf since the new `variant` field helps deduce whether we're dealing with original or thumbnail (or in the future specific image versions).